src/settings.rs

```diff
@@ -61,56 +62,85 @@ impl SettingsFile {
     }
 }
 
-#[derive(Clone, Default)]
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
 pub struct Settings {
     pub default_toolchain: Option<String>,
 }
 
-fn take_value(table: &mut Table, key: &str) -> Result<Value> {
-    if let Some(i) = table.remove(key) {
-        i.into_value()
-            .or_else(|_| bail!("Item is None for key: '{}'", key))
-    } else {
-        bail!("missing key: '{}'", key)
+impl Settings {
+    pub(crate) fn parse(toml: &str) -> Result<Self> {
+        let settings: Settings = de::from_str(toml)?;
+        Ok(settings)
     }
-}
 
-fn get_opt_string(table: &mut Table, key: &str) -> Result<Option<String>> {
-    if let Ok(v) = take_value(table, key) {
-        if let Value::String(s) = v {
-            Ok(Some(s.to_string()))
-        } else {
-            bail!("Expected string, got {}", key)
-        }
-    } else {
-        Ok(None)
+    pub(crate) fn stringify(self) -> Result<String> {
```

Comment:
`to_string` would be a little more conventional, or even `to_toml_string`

Reply:
I changed it to `to_string` here. Also noticed (thanks to clippy) since `ser::to_document` takes a reference to self, we can therefore change `into_toml` to `to_toml` instead